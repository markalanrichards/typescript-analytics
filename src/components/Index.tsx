import * as React from "react";

export interface IndexProps { compiler: string; framework: string; }

export const Index = (props: IndexProps) => <h1>Hello from {props.compiler} and {props.framework}!</h1>;
