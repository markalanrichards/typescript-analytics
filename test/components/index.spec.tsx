import * as React from "react"
import { configure, shallow, ShallowWrapper, HTMLAttributes } from "enzyme"
import Adapter from "enzyme-adapter-react-16"
import toJson from "enzyme-to-json"

configure({
  adapter: new Adapter()
})

import { Index } from "../../src/components/Index"

let index

beforeEach(
  () => (index = shallow(<Index compiler="TypeScript" framework="React" />))
)

it("should render without error", () => expect(index.length).toBe(1))

it("should render paragraph for each item that has been passed through props", () => {
  const h1Nodes: ShallowWrapper<HTMLAttributes, undefined> = index.find("h1")
  expect(h1Nodes.length).toBe({ items: [1] }.items.length)
})

it("should render paragraph for each item that has been passed through props", () => {
  const h1Nodes: ShallowWrapper<HTMLAttributes, undefined> = index.find("h1")
  expect(toJson(h1Nodes)).toMatchSnapshot()
})
