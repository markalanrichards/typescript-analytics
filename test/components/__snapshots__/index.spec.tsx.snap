// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render paragraph for each item that has been passed through props 1`] = `
<h1>
  Hello from 
  TypeScript
   and 
  React
  !
</h1>
`;
